server/firebase_setup.py

```diff
@@ -7,19 +7,27 @@
 
 def firebase_init():
     service_account_base64 = os.getenv('FIREBASE_SERVICE_ACCOUNT')
+    cred = None
```

Comment:
changes in this file just make sure that server starts when firebase file is not available right? why are we not using the dev mode variable here directly instead?

iirc firebase auth is also used to fetch stuff from firestore elsewhere, that should also be gated